propertyestimator/tests/test_simulation_layer.py

```diff
@@ -0,0 +1,191 @@
+"""
+Units tests for propertyestimator.layers.simulation
+"""
+
+import uuid
+from collections import OrderedDict
+
+import pytest
+from simtk import unit
+
+from propertyestimator.client import PropertyEstimatorOptions
+from propertyestimator.layers.simulation import DirectCalculation, DirectCalculationGraph
+from propertyestimator.properties import PropertyPhase
+from propertyestimator.properties.density import Density
+from propertyestimator.properties.dielectric import DielectricConstant
+from propertyestimator.properties.plugins import registered_properties
+from propertyestimator.substances import Mixture
+from propertyestimator.thermodynamics import ThermodynamicState
+from propertyestimator.utils import get_data_filename, graph
+from propertyestimator.workflow import WorkflowSchema
+
+
+def create_dummy_property(property_class):
+
+    substance = Mixture()
+    substance.add_component('C', 0.5)
+    substance.add_component('CO', 0.5)
+
+    dummy_property = property_class(thermodynamic_state=ThermodynamicState(temperature=298 * unit.kelvin,
+                                                                          pressure=1 * unit.atmosphere),
+                                    phase=PropertyPhase.Liquid,
+                                    substance=substance,
+                                    value=10 * unit.gram,
+                                    uncertainty=1 * unit.gram,
+                                    id=str(uuid.uuid4()))
+
+    return dummy_property
+
+
+@pytest.mark.parametrize("registered_property_name", registered_properties)
+def test_calculation_schema(registered_property_name):
+    """Tests serialisation and deserialization of a calculation schema."""
+
+    registered_property = registered_properties[registered_property_name]
+
+    schema = registered_property.get_default_calculation_schema()
+    schema.validate_interfaces()
+
+    json_schema = schema.json()
+    print(json_schema)
```

Comment:
I generally try to avoid print statements in tests. They can make it tough to find errors/parse verbose pytest output.

Reply:
In most cases I completely here, although here I've found it to be really helpful as when things fail, I can simply check the JSON to see what went wrong!